src/backend/distributed/shared_library_init.c

```diff
@@ -2275,6 +2275,16 @@ RegisterCitusConfigVariables(void)
 		GUC_STANDARD,
 		NULL, NULL, NULL);
 
+	DefineCustomIntVariable(
+		"citus.max_shard_size",
+		gettext_noop("Sets the max size of a Shard"),
+		NULL,
+		&MaxShardSize,
+		104857600, 102400, INT32_MAX,
```

Comment:
is the unit here kilobytes? 

That would make sense, since most size settings in PG are kilobytes.

would be good to use `GUC_UNIT_KB | GUC_STANDARD` as flags below

also, we should multiply by 1024 when comparing to the shard sizes, which are in bytes